symengine/logic.cpp

```diff
@@ -448,24 +448,43 @@ RCP<const Boolean> logical_xor(const vec_boolean &s)
             args.insert(a);
         }
     }
+    int nots = 0;
+    bool cantsimplify = false;
 
-    if (args.size() == 0) {
-        return boolFalse;
-    } else if (args.size() == 1) {
-        return *args.begin();
-    } else {
+    while (args.size() > 1 and not cantsimplify) {
+        cantsimplify = true;
         for (auto &a : args) {
             if (is_a<BooleanAtom>(*a)) {
                 SYMENGINE_ASSERT(a == boolTrue)
                 args.erase(a);
```

Comment:
You are erasing from the vector while iterating it. Segfaults can happen if you do that.